{{cookiecutter.project_slug}}/{{cookiecutter.project_slug}}/users/views.py

```diff
@@ -25,7 +25,7 @@ class UserUpdateView(LoginRequiredMixin, SuccessMessageMixin, UpdateView):
     success_message = _("Information successfully updated")
 
     def get_success_url(self):
-        return reverse("users:detail", kwargs={"username": self.request.user.username})
+        return self.request.user.get_absolute_url()  # type: ignore [union-attr]
```

Comment:
Can we avoid this `# type: ignore` by any chance? I would expect it to work if we add `-> str` annotation to the function definition at the line above. Is my expectation wrong?

Reply:
Unfortunately, there is no easy way to avoid this. The issue is that mypy doesn't know that the user is authenticated, which is something we know because `LoginRequiredMixin` runs first. One can do the following

> ```
> from django.http import HttpRequest
> from my_user_app.models import MyUser
> 
> class AuthenticatedHttpRequest(HttpRequest):
>     user: MyUser
> 
> ```
> And then use AuthenticatedHttpRequest instead of the standard HttpRequest for when you know that the user is authenticated. 

 as mentioned in the docs but that would make this view rather messy, in my opinion. Adding that `# type ...` seems like an easy fix for now.